src/operators/math/tensor/core.cairo

```diff
@@ -45,6 +46,7 @@ trait TensorTrait<T> {
     fn transpose(self: @Tensor<T>, axes: Span<usize>) -> Tensor<T>;
     fn reduce_sum(self: @Tensor<T>, axis: usize) -> Tensor<T>;
     fn argmax(self: @Tensor<T>, axis: usize) -> Tensor<usize>;
+    fn matmul(self: @Tensor<T>, other: @Tensor<T>) -> Tensor<T>;
```

Comment:
I would rename the function to just `mul` to preserve the naming and interface for when we support any ND array. Taking as reference how other libraries perform this task we can see:

- Numpy: uses the `matmul` naming.
- Torch: uses `mul` naming.
- ONNX: uses `mul` naming.

I would try to comply with the naming of ONNX for preserving the interface.

Reply:
Correct! I will change it!